baselines/flwr_baselines/publications/leaf/femnist/femnist/model.py

```diff
@@ -0,0 +1,174 @@
+"""Implementation of the model used for the FEMNIST experiments."""
+from logging import INFO
+from typing import Optional, Tuple
+
+import torch
+import torch.nn as nn
+from flwr.common.logger import log
+from torch.utils.data import DataLoader
+
+
+class Net(nn.Module):
+    """Implementation of the model used in the LEAF paper for training on
+    FEMNIST data."""
+
+    # pylint: disable=too-many-instance-attributes
+    def __init__(self, num_classes):
+        super().__init__()
+        self.conv1 = nn.Conv2d(1, 32, kernel_size=5, padding="same")
+        self.relu1 = nn.ReLU()
+        self.pool1 = nn.MaxPool2d(kernel_size=2, stride=2)
+        self.conv2 = nn.Conv2d(32, 64, kernel_size=5, padding="same")
+        self.relu2 = nn.ReLU()
+        self.pool2 = nn.MaxPool2d(kernel_size=2, stride=2)
+        self.fc1 = nn.Linear(7 * 7 * 64, 2048)
+        self.relu3 = nn.ReLU()
+        self.fc2 = nn.Linear(2048, num_classes)
+
+    def forward(self, x):
+        """Forward step in training."""
+        x = self.conv1(x)
+        x = self.relu1(x)
+        x = self.pool1(x)
+        x = self.conv2(x)
+        x = self.relu2(x)
+        x = self.pool2(x)
+        x = x.view(-1, 7 * 7 * 64)
+        x = self.fc1(x)
+        x = self.relu3(x)
+        x = self.fc2(x)
+        return x
+
+
+# pylint: disable=too-many-arguments, too-many-locals, no-member
+def train(
+    net: nn.Module,
+    trainloader: DataLoader,
+    valloader: DataLoader,
+    epochs: Optional[int],
+    learning_rate: float,
+    device: torch.device,
+    n_batches: Optional[int] = None,
+    verbose: bool = False,
+) -> Tuple[float, float, Optional[float], Optional[float]]:
+    """Train a given model with CrossEntropy and SGD (or some version of it
+    like batch-SGD).
+
+    n_batches is an alternative way of specifying the training length
+    (instead of epochs)
+    """
+    criterion = torch.nn.CrossEntropyLoss(reduction="sum")
+    optimizer = torch.optim.SGD(net.parameters(), lr=learning_rate)
+    net.train()
+    epoch_loss, epoch_acc = 0.0, 0.0
+    # pylint: disable=R1705
```

Comment:
This can be removed right? Because you are returning in both cases? (I am not sure)

Reply:
Generally, It says I should drop the "else", so I tried to write it in elif and add else that raises Error when both arguments are None but that didn't help